src/cpp/core/http/Response.cpp

```diff
@@ -46,13 +46,13 @@ namespace {
 #define kDeflateWindow 15
 #define kDefaultMemoryUsage 8
 
-boost::shared_ptr<StreamBuffer> makePaddingBuffer(size_t numPadding)
+std::shared_ptr<StreamBuffer> makePaddingBuffer(size_t numPadding)
```

Comment:
I think this will be quite a substantial change on top of the already substantial changes in this PR. Perhaps it's something we can look at doing later.

Reply:
Agree with @MariaSemple, this change is already very ambitious so let's not let anything else creep into scope! (pun intended) I'm assuming here that the compiler will stop us from mixing incompatible types so we don't need to worry about accidentally crossing these wires.

Reply:
I propose that after this is merged, we create an issue wherein we explore moving from `boost::shared_ptr<>` to `std::shared_ptr<>` throughout. (AFAIK Boost APIs are good about being smart-pointer-implementation-agnostic so it shouldn't cause too much pain)